woocommerce/woocommerce-admin
# Onboarding: Add setup theme step to profiler

Fixes #2707

Adds a theme setup step to activate themes and house logic for redirecting to wccom.

* Fixes theme update and profile item completion state bug.
* Fixes the `lastRequested` time for profile items updates.
* Adds in a “setup” screen to handle redirects and theme activation.

Note that actual theme activation APIs will be handled in https://github.com/woocommerce/woocommerce-admin/issues/3123

### Screenshots
<img width="369" alt="Screen Shot 2019-10-29 at 8 37 14 PM" src="https://user-images.githubusercontent.com/10561050/67768055-add08480-fa8c-11e9-93bd-e07be96a6f11.png">

### Detailed test instructions:

1. Go through the profiler, selecting options that will require purchase (product types and/or a non-installed theme).
2. Make sure the last step is a blank screen and redirects to the wccom checkout flow.
3.  Re-enable the profiler and walk through the steps selecting options that don't require purchase and selecting the currently active theme.
4. Note that the setup step is skipped and the profiler is complete since the theme is already active.
5. Re-enable the profiler one last time and don't select any product types that require purchases.
6. Before selecting a theme in your console add `wcSettings.onboarding.activeTheme = 'not-your-theme-name';`
7. Select your currently active theme.
8. Note styling for the setup step matches designs.
